src/rust/runtime/scheduler/scheduler.rs

```diff
@@ -162,8 +162,14 @@ impl Scheduler {
         assert!(self.current_running_task.is_some());
         let result: Option<Box<dyn Task>> = group.poll_notified_task_and_remove_if_ready(self.current_task_id);
         assert!(self.current_running_task.is_some());
-        *self.current_running_task = None;
+        let task_id: TaskId = self
+            .current_running_task
+            .take()
+            .expect("must have a current running task");
         assert!(self.current_running_task.is_none());
+        if result.is_some() {
+            self.ids.remove(&task_id);
```

Comment:
We are ignoring the return value of `result`. Could you please `assert!()` that the return value was `Some()`?

Reply:
We are not ignoring it. It gets passed back as the result of the function.

Reply:
Ah got it.